plugins/catalog-backend/src/database/DefaultProviderDatabase.ts

```diff
@@ -161,6 +161,11 @@ export class DefaultProviderDatabase implements ProviderDatabase {
             });
           }
 
+          await tx<DbRefreshStateReferencesRow>('refresh_state_references')
+            .where('target_entity_ref', entityRef)
+            .andWhere({ source_key: options.sourceKey })
+            .delete();
```

Comment:
This is the fix for the original issue, which lines up the logic with the equivalent action in the processing database. I imagine we'll likely look to introduce a unique constraint here too though.

Reply:
Will this handle cases where sourceKey is null?

Reply:
This table has two different sets of rows, source_key + target, and source_ref + target. The source_key is for references from entity providers to entities, while the source_ref is references from one entity to another. This code only deals with the source_key + target rows, as it's only encapsulating the operations used by entity providers.

Reply:
sorry. I was thinking `source_entity_ref`. Nvm